lib/firebase_functions_interop.dart

```diff
@@ -602,3 +606,101 @@ class CustomerEncryption {
 
   CustomerEncryption({this.encryptionAlgorithm, this.keySha256});
 }
+
+/// Namespace for Firebase Authentication functions.
+class AuthFunctions {
+  const AuthFunctions._();
+
+  /// Registers a Cloud Function to handle user authentication events.
+  UserBuilder user() => new UserBuilder._(_js.auth.user());
+}
+
+/// The Firebase Authentication user builder interface.
+class UserBuilder {
+  @protected
+  final js.UserBuilder nativeInstance;
+
+  UserBuilder._(this.nativeInstance);
+
+  /// Event handler that fires every time a Firebase Authentication user is created.
+  js.CloudFunction onCreate(FutureOr<void> handler(AuthEvent event)) {
+    dynamic wrapper(js.Event jsEvent) => _handleEvent(jsEvent, handler);
+    return nativeInstance.onCreate(allowInterop(wrapper));
+  }
+
+  /// Event handler that fires every time a Firebase Authentication user is deleted.
+  js.CloudFunction onDelete(FutureOr<void> handler(AuthEvent event)) {
+    dynamic wrapper(js.Event jsEvent) => _handleEvent(jsEvent, handler);
+    return nativeInstance.onDelete(allowInterop(wrapper));
+  }
+
+  dynamic _handleEvent(
+      js.Event jsEvent, FutureOr<void> handler(AuthEvent event)) {
+    final AuthEvent event = new AuthEvent(
+      data: new UserRecord(jsEvent.data),
+      eventId: jsEvent.eventId,
+      eventType: jsEvent.eventType,
+      params: dartify(jsEvent.params),
+      resource: jsEvent.resource,
+      timestamp: DateTime.parse(jsEvent.timestamp),
+    );
+    var result = handler(event);
+    if (result is Future) {
+      return futureToPromise(result);
+    }
+    return null;
+  }
+}
+
+/// Interface representing a user.
+class UserRecord {
+  UserRecord(js.UserRecord this.nativeInstance);
+
+  @protected
+  final js.UserRecord nativeInstance;
+
+  /// Whether or not the user is disabled.
+  bool get disabled => nativeInstance.disabled;
+
+  /// The user's display name.
+  String get displayName => nativeInstance.displayName;
+
+  /// The user's primary email, if set.
+  String get email => nativeInstance.email;
+
+  /// Whether or not the user's primary email is verified.
+  bool get emailVerified => nativeInstance.emailVerified;
+
+  /// Additional metadata about the user.
+  UserMetadata get metadata => nativeInstance.metadata;
+
+  /// The user's photo URL.
+  String get photoURL => nativeInstance.photoURL;
+
+  /// An array of providers (for example, Google, Facebook) linked to the user.
+  List<UserInfo> get providerData => nativeInstance.providerData;
```

Comment:
I was not sure if JS array maps nicely to Dart's `List` (e.g. some methods may not work, like `List.map` or `List.fold`). Just ran some tests and it seems working fine.